frontend/ui/lib/model/place_visit.dart

```diff
@@ -8,7 +8,7 @@ enum UserMark {
 }
 
 class PlaceVisit {
-  UserMark userMarkEnumFromString(String userMark) {
+  static UserMark userMarkEnumFromString(String userMark) {
```

Comment:
Since you're fixing this here anyway, please precompute the `String -> UserMark` mapping and store it in a static map. That also allows you to explicitly check for a matching string and throw a specific exception on bad input (rather than whatever `firstWhere` throws in this case).